soc/xtensa/nxp_adsp/common/interrupt.c

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (c) 2021 NXP Semiconductors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Daniel Baluta <daniel.baluta@nxp.com>
+ * Author: Paul Olaru <paul.olaru@nxp.com>
+ */
+
+#include <zephyr.h>
+#include <spinlock.h>
+
+#include <zephyr/types.h>
+#include <arch/xtensa/arch.h>
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "interrupt.h"
+
+/*
+ * The IRQ_STEER module takes 512 shared interrupts and delivers them
+ * via 8 interrupt lines to any other component. It supports 5 channels,
+ * one of them being for the DSP itself (channel 0).
```

Comment:
You might want to consider moving this code into drivers/interrupt_controller.  That's the way other SOCs do it, though I don't have strong feelings (an interrupt controller is never going to be a "normal" device, it's very low level soc/arch code by definition).

Reply:
I've removed it.
We'll use the irq_steer from SOF, for now.